src/main/java/com/qunite/api/service/QueueServiceImpl.java

```diff
@@ -48,6 +49,27 @@ public Optional<Integer> getMemberPositionInQueue(Long memberId, Long queueId) {
         .map(entry -> entry.getEntryIndex() + 1);
   }
 
+  @Override
+  @Transactional(isolation = Isolation.REPEATABLE_READ)
+  public void changeMemberPositionInQueue(Long memberId, Long queueId, Integer newIndex) {
+    entryRepository.findById(new EntryId(memberId, queueId)).ifPresent(entry -> {
+      var currentIndex = entry.getEntryIndex();
+      if (!currentIndex.equals(newIndex)) {
+        var startIndex = Math.min(currentIndex, newIndex);
+        var endIndex = Math.max(currentIndex, newIndex);
+        var increment = currentIndex < newIndex ? -1 : 1;
+
+        entryRepository.updateEntryIndexes(queueId, startIndex, endIndex, increment);
+        entry.setEntryIndex(newIndex);
+      }
+    });
+  }
```

Comment:
I feel myself puzzled about opportunity to "jump forward" along the queue. Currenty, if I have, let`s say, list of entries like this:
Entry index: 0, EntryId(memberId=7, queueId=1)
Entry index: 1, EntryId(memberId=6, queueId=1)
Entry index: 2, EntryId(memberId=5, queueId=1)
Entry index: 3, EntryId(memberId=4, queueId=1)
Entry index: 4, EntryId(memberId=3, queueId=1)

I have an opportunity to change member`s (with id=3) entry index from 4 to, for example, 1:
AFTER
Entry index: 0, EntryId(memberId=7, queueId=1)
Entry index: 1, EntryId(memberId=3, queueId=1)
Entry index: 2, EntryId(memberId=6, queueId=1)
Entry index: 3, EntryId(memberId=5, queueId=1)
Entry index: 4, EntryId(memberId=4, queueId=1)

Honestly speaking, I don`t see scenarios for such a behaviour and this feature of method can violate users order in queue, isn`t it?